src/redux/reducers/postsSlice.js

```diff
@@ -0,0 +1,33 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import { getPosts } from "../../services";
+
+const initialState = {
+    status: "idle",
+    error: null,
+    posts: []
+};
+
+export const loadPosts = createAsyncThunk('posts/loadPosts',()=> getPosts({}));
+const postSlice = createSlice({
+    name: 'posts',
+    initialState,
+    reducers: {
+        updatePosts: (state, action) => {
+            state.posts = action.payload;
+        }
+    },
+    extraReducers: {
+        [loadPosts.pending]: (state) => {
+            state.status = 'loading';
+        },
+        [loadPosts.fulfilled]: (state, action) => {
+            console.log("enter in fulfilled", action.payload);
+            state.status = 'fulfilled';
+            state.posts = action.payload;
+        }
+    }
+})
+
+export default postSlice.reducer;
+export const {updatePosts} = postSlice.actions;
```

Comment:
Currently, you only have 1 slice file, so this is fine, when you will have multiple slice files, it would be more manageable, if you export actions from slice files and then barrel export it from 1 common file, just suggesting for future reference.

Reply:
Thanks for the suggestion, I will keep it in mind for later